packages/graph/hash_graph/postgres_migrations/V1__initial.sql

```diff
@@ -0,0 +1,270 @@
+  ADD CONSTRAINT "entity_ids_left_reference" FOREIGN KEY ("left_owned_by_id", "left_entity_uuid") REFERENCES "entity_ids",
+  ADD CONSTRAINT "entity_ids_right_reference" FOREIGN KEY ("right_owned_by_id", "right_entity_uuid") REFERENCES "entity_ids",
+  ADD CONSTRAINT "entity_ids_relation_constraint" CHECK (left_entity_uuid IS NULL AND right_entity_uuid IS NULL AND left_owned_by_id IS NULL AND right_owned_by_id IS NULL     OR        left_entity_uuid IS NOT NULL AND right_entity_uuid IS NOT NULL AND left_owned_by_id IS NOT NULL AND right_owned_by_id IS NOT NULL);
+
+CREATE TABLE IF NOT EXISTS "entity_editions" (
+  "entity_record_id" BIGINT PRIMARY KEY NOT NULL GENERATED ALWAYS AS IDENTITY,
+  "entity_type_version_id" UUID NOT NULL REFERENCES "entity_types",
+  "properties" JSONB NOT NULL,
+  "left_to_right_order" integer,
+  "right_to_left_order" integer,
+  "updated_by_id" UUID NOT NULL REFERENCES "accounts",
+  "archived" BOOLEAN NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS "entity_versions" (
+  "owned_by_id" UUID NOT NULL,
+  "entity_uuid" UUID NOT NULL,
+  "entity_record_id" BIGINT NOT NULL REFERENCES "entity_editions",
+  "decision_time" tstzrange NOT NULL,
+  "transaction_time" tstzrange NOT NULL
+);
+
+ALTER TABLE "entity_versions"
+  ADD CONSTRAINT "entity_versions_reference" FOREIGN KEY ("owned_by_id", "entity_uuid") REFERENCES "entity_ids",
+  ADD CONSTRAINT "entity_versions_overlapping" EXCLUDE USING gist (owned_by_id WITH =, entity_uuid WITH =, decision_time WITH &&, transaction_time WITH &&) DEFERRABLE INITIALLY IMMEDIATE,
+  ADD CONSTRAINT "entity_versions_decision_time_validation" CHECK (lower(decision_time) <= lower(transaction_time));
+
+CREATE VIEW "entities" AS 
+    SELECT
+      entity_versions.entity_record_id,
+      entity_versions.owned_by_id,
+      entity_versions.entity_uuid,
+      entity_versions.decision_time,
+      entity_versions.transaction_time,
+      entity_editions.entity_type_version_id,
+      entity_editions.updated_by_id,
+      entity_editions.properties,
+      entity_editions.archived,
+      entity_ids.left_owned_by_id,
+      entity_ids.left_entity_uuid,
+      entity_editions.left_to_right_order,
+      entity_ids.right_owned_by_id,
+      entity_ids.right_entity_uuid,
+      entity_editions.right_to_left_order
+    FROM entity_versions
+    JOIN entity_editions ON entity_versions.entity_record_id = entity_editions.entity_record_id
+    JOIN entity_ids ON entity_versions.owned_by_id = entity_ids.owned_by_id AND entity_versions.entity_uuid = entity_ids.entity_uuid;
+
+CREATE OR REPLACE FUNCTION "create_entity"("_owned_by_id" UUID, "_entity_uuid" UUID, "_decision_time" TIMESTAMP WITH TIME ZONE, "_updated_by_id" UUID, "_archived" BOOLEAN, "_entity_type_version_id" UUID, "_properties" JSONB, "_left_owned_by_id" UUID, "_left_entity_uuid" UUID, "_right_owned_by_id" UUID, "_right_entity_uuid" UUID, "_left_to_right_order" INTEGER, "_right_to_left_order" INTEGER)
```

Comment:
Does Postgres allow linebreaks here? This would be more readable.

Reply:
3044e5dbf1116c098d3f00b7325dd0623d5d488d